meson.build

```diff
@@ -3,7 +3,7 @@ project(
   'c',
   version: '0.8.2',
   license: 'GPL-2.0-only',
-  meson_version: '>=0.59.0',
+  meson_version: '>=1.1',
```

Comment:
Debian Stable (and this likely a bunch of other distros using it as a base) is still on meson 1.0.1: https://tracker.debian.org/pkg/meson

As a result I am a bit skeptical about this change at the current point in time as it isn't really needed from the technical side. wlroots 0.18.x also still uses `meson_options.txt`.